pkg/controllers/nodeclaim/lifecycle/registration.go

```diff
@@ -121,11 +120,16 @@ func (r *Registration) syncNode(ctx context.Context, nodeClaim *v1.NodeClaim, no
 	controllerutil.AddFinalizer(node, v1.TerminationFinalizer)
 
 	node = nodeclaimutils.UpdateNodeOwnerReferences(nodeClaim, node)
-	node.Labels = lo.Assign(node.Labels, nodeClaim.Labels)
+
+	// We do not sync the taints if the feature flag is enabled. We instead assume that the karpenter provider
+	// is managing taints. We still manage/remove the unregistered taint to signal the end of syncing.
+	if _, ok := node.Annotations[v1.NodeDoNotSyncTaintsLabelKey]; !ok {
```

Comment:
We need to make sure that value is label and not an annotation. The reason for this is that kubelet configuration does not allow us to add annotation, but will allow us to register nodes with labels.